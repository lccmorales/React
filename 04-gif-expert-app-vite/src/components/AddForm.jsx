import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddForm = ( { setCategories, setLimitNumGifs } ) => {
    const [inputValueSearch, setInputValueSearch] = useState('');
    const [inputValueLimit, setInputValueLimit] = useState('');

    const handleInputSearchChange = (e) => {
        setInputValueSearch(e.target.value);
        console.log('handleInputSearchChange - llamado');
    }

    const handleInputLimitChange = (e) => {
        setInputValueLimit(e.target.value);
        console.log('handleInputLimitChange - llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('handleSubmit - llamado', inputValueSearch);
        if (inputValueSearch.trim().length > 3 && parseInt(inputValueLimit.trim()) > 0) {
            setCategories( cats => [ inputValueSearch, ...cats,] );
            setInputValueSearch('');
            setLimitNumGifs( parseInt(inputValueLimit) );
            setInputValueLimit('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValueSearch }</p>
            <p>{ inputValueLimit }</p>
            <h3>Busqueda</h3>
            <input
                className='search'
                type="text"
                value={ inputValueSearch }
                onChange={ handleInputSearchChange }
            />

            <h3>Cantidad de Gif's a mostrar</h3>
            <input
                type="number"
                value={ inputValueLimit }
                onChange={ handleInputLimitChange }
            />

            <button type="submit">Buscar</button>
        </form>
    )
}

AddForm.propTypes = {
    setCategories: PropTypes.func.isRequired,
    setLimitNumGifs: PropTypes.func.isRequired
};