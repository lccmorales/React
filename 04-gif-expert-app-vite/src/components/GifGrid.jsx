import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( { category, limitNumGifs } ) => {
    const { images, loading } = useFetchGifs( category, limitNumGifs );

    return (
        <>
        <h3 className='animate__animated animate__flash' aria-label='titulo'>{ category }</h3>

        { loading && <p className='animate__animated animate__flash'>Cargando...</p> }
        
        <div className='card-grid'>
            {
                images.map( img => (
                    <GifGridItem
                        key={ img.id }
                        { ...img }
                    />
                ))
            }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    limitNumGifs: PropTypes.number.isRequired
}