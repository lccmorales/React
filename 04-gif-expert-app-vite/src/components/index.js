export * from './AddForm';
export * from './GifGrid';
export * from './GifGridItem';