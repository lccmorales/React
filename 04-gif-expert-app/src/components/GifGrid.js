import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types';

export const GifGrid = ( {category} ) => {

    const [count, setCount] = useState(0);

    useEffect( () => {
        getGifs();
    }, []);

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Arcane&limit=10&api_key=wGFpRizoH8FkJI4Dbmue2wSoctrFUqGq';
        const resp = await fetch ( url );
        const { data } = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        console.log(gifs);
    }

    return (
        <>
            <h3>{ category }</h3>
            <h1>{ count }</h1>
            <button onClick = { () => setCount(count + 1) }>+1</button>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}