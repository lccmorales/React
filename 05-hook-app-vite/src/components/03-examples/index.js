export * from './Button';
export * from './LoadingQuote';
export * from './MultipleCustomHooks';
export * from './Quote';