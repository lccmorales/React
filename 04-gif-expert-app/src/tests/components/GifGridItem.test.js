import React from 'react';
// import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en GifGridItem.js', () => {
    test('debe de mostrar <GifGridItem /> correctamente', () => {
        const id = 'PM7EBVwNHRuXYKIsJf';
        const img = {
            id: id,
            title: 'Dance Jinx GIF LOL',
            url: 'https://media0.giphy.com/media/PM7EBVwNHRuXYKIsJf/giphy-downsized-medium.gif?cid=96007aa4ac1qtu0yql3o5v70s2q3y3v7g3erl7sdkx2qbts0&rid=giphy-downsized-medium.gif&ct=g'
        };
        const wrapper =  shallow(<GifGridItem />);
        expect( wrapper ).toMatchSnapshot();
    })
})