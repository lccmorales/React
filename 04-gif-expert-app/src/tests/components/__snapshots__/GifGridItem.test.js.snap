// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem.js debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__swing"
>
  <img
    alt="Dance Jinx GIF LOL"
    src="https://media0.giphy.com/media/PM7EBVwNHRuXYKIsJf/giphy-downsized-medium.gif?cid=96007aa4ac1qtu0yql3o5v70s2q3y3v7g3erl7sdkx2qbts0&rid=giphy-downsized-medium.gif&ct=g"
  />
  <p>
    Dance Jinx GIF LOL
  </p>
</div>
`;
